commands/build.go

```diff
@@ -495,6 +519,10 @@ func buildCmd(dockerCli command.Cli, rootOpts *rootOptions) *cobra.Command {
 
 	flags.Var(options.ulimits, "ulimit", "Ulimit options")
 
+	flags.StringArrayVar(&options.attests, "attest", []string{}, `Attestation parameters (format: "type=sbom,generator=image")`)
+	flags.StringVar(&options.sbom, "sbom", "", `Shorthand for "--attest=type=sbom"`)
+	flags.StringVar(&options.provenance, "provenance", "", `Shortand for "--attest=type=provenance"`)
```

Comment:
```console
$ docker buildx build --sbom .
ERROR: "docker buildx build" requires exactly 1 argument.
See 'docker buildx build --help'.

Usage:  docker buildx build [OPTIONS] PATH | URL | -

Start a build
```

I think you should just set a default value for `--sbom` to `type=sbom` and same for `--provenance` with `type=provenance` so you don't need to parse bool. Could be done with `NoOptDefVal`: https://github.com/spf13/pflag/#setting-no-option-default-values-for-flags

```go
flags.StringVar(&options.sbom, "sbom", "", `Shorthand for "--attest=type=sbom"`)
flags.Lookup("sbom").NoOptDefVal = "type=sbom"
```

Reply:
The boolean flag was added as @tonistiigi mentioned above, so I think we should keep that:

> An alternative would be --sbom --provenance , I guess the arguments would need to be [bool=true|csv]. Not sure how many more attestation types we will need in the future.

For the empty arg, cobra has an open issue that prevents `NoOptDefVal` from working without the equals, i.e. of the form `--sbom generator=x` instead of `--sbom=generator=x`: https://github.com/spf13/cobra/issues/866. Ideally we should support both, like we do with the exporters/cache backends.

Reply:
> The boolean flag was added as @tonistiigi mentioned above, so I think we should keep that:

SGTM

> For the empty arg, cobra has an open issue that prevents `NoOptDefVal` from working without the equals, i.e. of the form `--sbom generator=x` instead of `--sbom=generator=x`: [spf13/cobra#866](https://github.com/spf13/cobra/issues/866).

That's sad :disappointed:

Reply:
I'm not sure how cobra would differentiate from an `[]args` entry (the context path in our build commands) and a flag value with the `--flag value` format. Once you say the value is optional, it cobra needs something to differentiate the two.